pkg/registry/core/service/rest.go

```diff
@@ -586,6 +534,54 @@ func (rs *REST) initClusterIP(service *api.Service) (bool, error) {
 	return false, nil
 }
 
+func (rs *REST) initNodePort(service *api.Service, nodePortOp *portallocator.PortAllocationOperation) error {
+	assignNodePorts := shouldAssignNodePorts(service)
```

Comment:
Thanks @freehan !

In fact, I thought to remove shouldAssignNodePorts here because we could assume the caller of `initNodePort` should guarantee that only the NodePort/LoadBalancer service can invoke `initNodePort` (see line 108). But I find it not easy to remove it due to the existing logic, so I leave it as-is. I will try to do more refactoring to remove it here. One small question: should I use line 108 or use `shouldAssignNodePorts` as the checking conditions outside of initNodePort? I prefer to line 108 because it seems to be more explicit. WDYT?